drivers/SmartThings/sonos/src/init.lua

```diff
@@ -29,7 +31,7 @@ local function find_player_for_device(driver, device, should_continue)
     local attempts = 0
     should_continue = function()
       attempts = attempts + 1
-      return attempts <= 10
+      return attempts <= DEFAULT_SSDP_RETRY_ATTEMPTS
```

Comment:
Whats the reasoning for limiting the discovery attempts? If the max attempts are hit, a device will be non-functional because it failed to initialize; even if the device came back online, it wouldn't initialize until the driver restarted. IMO an exponential backoff would be preferred, but that is a relatively substantial change to be making directly to beta now.

Reply:
Agreed, we should make a ticket and revisit this. Don't wanna make that change going in to beta.

Reply:
@dljsjr @ctowns and I discussed this and decided it is nice to have but because we have yet to see an issue in tests around this that it is not critical. We will re-evaluate after the first set of migrations.